### ISSUE TITLE
Render Issue

### ISSUE THREAD

**Issue by givemhell** [lumien231/Hardcore-Darkness#5]
it seems there is an issue with rendering forge microblocks at night
http://i.imgur.com/V9So2rT.png  <-- night 

http://i.imgur.com/MEKH6LW.png <-- day

this problem does not happen when hardcore darkness is removed from my mods folder

**Comment by givemhell** [comment]
@lumien231 do you plan on doing something about this? or at least acknowledging the problem?

**Comment by lumien231** [comment]
If i find out what causes it i will fix it :P.

**Comment by givemhell** [comment]
it also seems to affect carpender blocks, i would think its got something todo with how light is calculated. are you accounting for blocks like slabs?

**Comment by lumien231** [comment]
You see i'm not accounting for anything^^. I really am just editing the vanilla lightmap to not have that minimal light.

**Comment by givemhell** [comment]
lets see what CptSpaceToaster has to say on this, he should be able to shed some light on this,
if your feeling adventuress take a look here & see if this helps you get any ideas on how you might go about solving the problem

https://github.com/CptSpaceToaster/CptsModdingLight/issues/36

**Comment by lumien231** [comment]
Considering this happens with FMP and not Colored Lights or Dynamic Lights or whatever i'm not sure how that issue is related?

**Comment by givemhell** [comment]
this should make more sense, my bad im really bad at explaining things >.<

Colored Lights core changes vanilla light system
Hardcore Darkness hooks the vanilla light system thats been changed 
Blocks from random mods render with a shade of blue during the night

**Comment by lumien231** [comment]
Lulz, you should have maybe mentioned Colored Lights Core considering THAT is the mod that is causing this together with Hardcore Darkness.

**Comment by givemhell** [comment]
yeah i swore that i removed it during the test but i guess i removed it from the wrong pack,

**Comment by CptSpaceToaster** [comment]
So, I replaced the lighting engine with my own, but it's backwards compatible... I don't think that's being discussed here.  To make sure we're on the same page:  

1.) Does hardcore darkness mess with the lighting engine in such a way, that light spreads differently?  Does Sunlight still spread around?
2.) Does block light still flood-out like vanilla?  (y/n)

I don't care how it's DRAWN quite yet... but that will help diagnose what's going wrong here.

The second thing I did, was replace the lightmap with a shader......  This change is not 100% backwards compatible, and quite a bother.  [This](https://gist.github.com/CptSpaceToaster/8b6dbd107e5fe44ab7b7) document details the change I made, and can help explain why things turn blue (in format 2)

3.) What does Hardcore darkness change in the rendering pipeline?  If it's strictly editing the lightmap... it's going to be successful, but the change won't have much meaning... I guess I could hack in a check for hardcore darkness, and then disable what I do... which is kinda hilarious... but it would work.

If you're too busy to respond, that's fine, I'll go mess with your source if I need to.  Right now, I've got other things to fix, but this is on the list.  A couple people have asked for it.

**Comment by lumien231** [comment]
I'm just removing all of the "*0.8F+0.2F" type of calculations in getSunBrightnessBody and EntityRenderer.updateLightmap, i don't change anything else.

**Comment by CptSpaceToaster** [comment]
yep... that logic will simply be thrown out the window... I'll have to figure something out on my end.
